Add parser for extended floating point numbers

st-pasha opened the pull request:
Closes #1800

mattdowle commented:
NB: fails are due to knitr/R-devel unrelated to this PR.  master has the same problem will have to look at separately.

st-pasha commented:
Fixed the parser so that it returns `NAND` as before.
Added new issue #2363 to make sure that we fix the `test()` function at some point so that it can catch problems like these.

mattdowle commented:
After adding the stricter `identical`, one of them (1830.6) fails with a NA/NaN difference to be sorted out.
